src/utils/localStorageHelpers.js

```diff
@@ -0,0 +1,43 @@
+const LOCAL_STORAGE_KEY = 'modules';
+
+/**
+ * Get any existing modules that exist in local storage
+ *
+ * @returns Object keyed by module name whose values are the module JSON
+ */
+export function getLocalStorageModules() {
+  const existingModules = localStorage.getItem(LOCAL_STORAGE_KEY);
+
+  if (!existingModules) return {};
+
+  return JSON.parse(existingModules);
+}
+
+/**
+ * Save a module to local storage
+ *
+ * @param {Object} module the JSON content of the module being saved
+ */
+export function saveLocalStorageModule(module) {
+  const newModuleObject = { ...getLocalStorageModules() };
+
+  newModuleObject[module.name] = module;
```

Comment:
I noticed that if you open up the same module from multiple sources (ex, from Core Modules and from GitHub > master) the builder allows you to have 2 tabs, but if you save both to local storage you only get one. I'm not sure what the best approach here is - should we allow saving multiple instances of the "same" module? (for example, by using the tab title as the key rather than `module.name`) Or is it not worth it and we should just use module.name and not allow 2 instances of the same module to exist in localStorage simultaneously.
Thoughts?

Reply:
I'm open to using the tab title. What do you think makes the most sense from a user perspective?

Reply:
I added a confirm dialog to prompt the user to overwrite a module with the same name if it exists